llvm/lib/Target/DirectX/DXContainerGlobals.cpp

```diff
@@ -186,51 +186,71 @@ void DXContainerGlobals::addResourcesForPSV(Module &M, PSVRuntimeInfo &PSV) {
   DXILResourceTypeMap &DRTM =
       getAnalysis<DXILResourceTypeWrapperPass>().getResourceTypeMap();
 
-  for (const dxil::ResourceBindingInfo &RBI : DBM) {
+  auto MakeBinding =
+      [](const dxil::ResourceBindingInfo::ResourceBinding &Binding,
+         const dxbc::PSV::ResourceType Type, const dxil::ResourceKind Kind,
+         const dxbc::PSV::ResourceFlags Flags = dxbc::PSV::ResourceFlags()) {
+        dxbc::PSV::v2::ResourceBindInfo BindInfo;
+        BindInfo.Type = Type;
+        BindInfo.LowerBound = Binding.LowerBound;
+        BindInfo.UpperBound = Binding.LowerBound + Binding.Size - 1;
+        BindInfo.Space = Binding.Space;
+        BindInfo.Kind = static_cast<dxbc::PSV::ResourceKind>(Kind);
```

Comment:
Would this conversion be safer done with a helper function or constructor? I couldn't find the definition for dxbc::PSV::ResourceKind but dxil::ResourceKind was just an enum. Seems like something that we could accidentally break ?

Reply:
`dxbc::PSV::ResourceKind` and `dxil::ResourceKind` happen to be the exact same enum. One is defined in [DXContainerConstants](https://github.com/llvm/llvm-project/blob/llvmorg-21-init/llvm/include/llvm/BinaryFormat/DXContainerConstants.def#L168) and the other in [DXILABI](https://github.com/llvm/llvm-project/blob/llvmorg-21-init/llvm/include/llvm/Support/DXILABI.h#L34), but if they didn't match exactly we would have a serious problem.

This means that technically the static cast is safe - but it's definitely a break in abstraction. Arguably we should tie these two enums together in a clearer way, though I'm not sure exactly where that should best happen.

In any case, I think that should be out of scope of this PR - I am just moving this cast after all, not introducing it here.